File: src/Entity/Destination/Parkingspot.php
<?php

namespace App\Entity\Destination;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

#[ORM\Entity]
class Parkingspot
{

   /* #[ORM\ManyToOne(targetEntity: Destination::class)]
    #[ORM\JoinColumn(name: 'id_destination', referencedColumnName: 'id_destination')]
    private ?Destination $id_destination = null;*/

    #[ORM\ManyToOne(targetEntity: Destination::class, inversedBy: "parkingSpots")]
    #[ORM\JoinColumn(name: "id_destination", referencedColumnName: "id")]
    private ?Destination $id_destination = null;
    
    #[ORM\Id]
    #[ORM\GeneratedValue (strategy: "IDENTITY")]
    #[ORM\Column(type: 'integer')]
    private int $id_parking;

    #[ORM\Column(name: 'is_accessible',type: 'string')]
    #[Assert\NotBlank(message: "Accessibility field cannot be blank.")]
    #[Assert\Choice(choices: ["yes", "no"], message: "Accessibility must be 'yes' or 'no'.")]
    private string $isAccessible;

    #[ORM\Column(name:'is_available',type: 'string')]
    #[Assert\NotBlank(message: "Availability field cannot be blank.")]
    #[Assert\Choice(choices: ["yes", "no"], message: "Availability must be 'yes' or 'no'.")]
    private string $isAvailable;

    #[ORM\Column(name:'spot_number',type: 'string', length: 255)]
    #[Assert\NotBlank(message: "Spot number cannot be blank.")]
    #[Assert\Length(max: 255, maxMessage: "Spot number cannot be longer than 255 characters.")]
    private string $spotNumber;

    #[ORM\Column(name:'parking_fees',type: 'string')]
    #[Assert\NotBlank(message: "Parking fees cannot be blank.")]
    #[Assert\Regex(pattern: "/^\d+(\.\d{1,2})?$/", message: "Fees must be a valid number (e.g., 2.50).")]

    private string $parkingFees;

    #[ORM\Column(type: 'string')]
    #[Assert\NotBlank(message: "Size field cannot be blank.")]
    #[Assert\Regex(
        pattern: '/^\d+$/',
        message: "Parking fees must be a valid number."
    )]
    private string $size;

    #[ORM\Column(type: 'string', length: 65535)]
    #[Assert\NotBlank(message: "Notes cannot be blank.")]
    private string $notes;


    public function getId_parking(): ?int
    {
        return $this->id_parking;
    }

    public function setId_parking(int $id_parking): self
    {
        $this->id_parking = $id_parking;
        return $this;
    }

    public function getIsAccessible(): ?string
    {
        return $this->isAccessible;
    }

    public function setIsAccessible(string $isAccessible): self
    {
        $this->isAccessible = $isAccessible;
        return $this;
    }

    public function getIsAvailable(): ?string
    {
        return $this->isAvailable;
    }

    public function setIsAvailable(string $isAvailable): self
    {
        $this->isAvailable = $isAvailable;
        return $this;
    }

    public function getSpotNumber(): ?string
    {
        return $this->spotNumber;
    }

    public function setSpotNumber(string $spotNumber): self
    {
        $this->spotNumber = $spotNumber;
        return $this;
    }

    public function getParkingFees(): ?string
    {
        return $this->parkingFees;
    }

    public function setParkingFees(string $parkingFees): self
    {
        $this->parkingFees = $parkingFees;
        return $this;
    }

    public function getSize(): ?string
    {
        return $this->size;
    }

    public function setSize(string $size): self
    {
        $this->size = $size;
        return $this;
    }

    public function getNotes(): ?string
    {
        return $this->notes;
    }

    public function setNotes(string $notes): self
    {
        $this->notes = $notes;
        return $this;
    }

    public function getIdDestination(): ?Destination
{
    return $this->id_destination;
}


    public function setIdDestination(?Destination $id_destination): self
{
    $this->id_destination = $id_destination;
    return $this;
}

}

File: src/main/java/Models/ParkingSpot.java
package Models;

public class ParkingSpot {
    private int id_parking, size;
    private int id_destination;
    private boolean is_available ,is_accessible;
    private String spot_number,notes;
    private float parking_fees;

    public ParkingSpot(int id_parking,int id_destination, int size, boolean is_available, boolean is_accessible, String spot_number, String notes, float parking_fees) {
        this.id_destination=id_destination;
        this.id_parking = id_parking;
        this.size = size;
        this.is_available = is_available;
        this.is_accessible = is_accessible;
        this.spot_number = spot_number;
        this.notes = notes;
        this.parking_fees = parking_fees;
    }

    public ParkingSpot(int id_destination,int size, boolean is_available, boolean is_accessible, String spot_number, String notes, float parking_fees) {
        this.id_destination=id_destination;
        this.size = size;
        this.is_available = is_available;
        this.is_accessible = is_accessible;
        this.spot_number = spot_number;
        this.notes = notes;
        this.parking_fees = parking_fees;
    }

    public int getId_parking() {
        return id_parking;
    }

    public void setId_parking(int id_parking) {
        this.id_parking = id_parking;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public boolean isIs_available() {
        return is_available;
    }

    public void setIs_available(boolean is_available) {
        this.is_available = is_available;
    }

    public boolean isIs_accessible() {
        return is_accessible;
    }

    public void setIs_accessible(boolean is_accessible) {
        this.is_accessible = is_accessible;
    }

    public String getSpot_number() {
        return spot_number;
    }

    public void setSpot_number(String spot_number) {
        this.spot_number = spot_number;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public float getParking_fees() {
        return parking_fees;
    }

    public void setParking_fees(float parking_fees) {
        this.parking_fees = parking_fees;
    }

    public int getId_destination() {
        return id_destination;
    }

    public void setId_destination(int id_destination) {
        this.id_destination = id_destination;
    }

    @Override
    public String toString() {
        return "ParkingSpot{" +
                "id_parking=" + id_parking +
                ", size=" + size +
                ", id_destination=" + id_destination +
                ", is_available=" + is_available +
                ", is_accessible=" + is_accessible +
                ", spot_number='" + spot_number + '\'' +
                ", notes='" + notes + '\'' +
                ", parking_fees=" + parking_fees +
                '}';
    }

}
